contracts/protocol/bases/GroupBase.sol

```diff
@@ -113,11 +120,32 @@ contract GroupBase is ProtocolBase, IBosonGroupEvents {
             valid = (_condition.tokenAddress == address(0) &&
                 _condition.tokenId == 0 &&
                 _condition.threshold == 0 &&
-                _condition.maxCommits == 0);
+                _condition.maxCommits == 0 &&
+                _condition.length == 0);
         } else if (_condition.method == EvaluationMethod.Threshold) {
-            valid = (_condition.tokenAddress != address(0) && _condition.maxCommits > 0 && _condition.threshold > 0);
-        } else if (_condition.method == EvaluationMethod.SpecificToken) {
-            valid = (_condition.tokenAddress != address(0) && _condition.threshold == 0 && _condition.maxCommits > 0);
+            valid = (_condition.tokenAddress != address(0) &&
+                _condition.maxCommits > 0 &&
+                _condition.threshold > 0 &&
+                _condition.length == 0 &&
+                _condition.tokenId == 0);
+        } else {
+            // SpecificToken
+            valid = (_condition.tokenAddress != address(0) &&
+                _condition.maxCommits > 0 &&
+                _condition.tokenType != TokenType.FungibleToken); // FungibleToken not allowed for SpecificToken
+
+            if (_condition.tokenId != 0) {
+                // SpecificToken with tokenId should have length
+                valid = valid && _condition.length > 0;
```

Comment:
We should also make sure that `length + tokenId` does not overflow, otherwise we can run into problems in `authorizeCommit`

Reply:
overflow check added